### PR TITLE
Toeplitz

### LINKED ISSUE #1231: Finding Matrix follows Teoplitz Matrix Algorithm or Not ?
Hi,
I am GSSOC'21 participant

My code will be in JAVA with proper comments and test cases.
@rudrakshi99 @nakul-19 @plazzy99 @tarun26091999  please assign me this issue

### PR DESCRIPTION
**Pull request by krishna-NIT**
Fixes #1231 
@tarun26091999 @yasharth291
Please Review it !

### CODE REVIEW

**Comment by krishna-NIT** [comment]
@rish-singhal  Please Review it sir !

File: 2D_Array/readme.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ Due to the fact that the elements of 2D arrays can be random accessed. Similar t
 ## Questions :
 * Spiral Print ----> [C++](/Code/C++/spiral_print.cpp)
 * Wave Print ----> [C++](/Code/C++/wave_print.cpp)
+* Checking Teoplitz Matrix Algorithm ----> [Java](/Code/Java/Toeplitz.java)
```
Comment (rish-singhal) [inline 2D_Array/readme.md:13]: Sort this list in lexographical order.

File: Code/Java/Toeplitz.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+A matrix is Toeplitz if every diagonal from top-left to bottom-right has the same elements.
+or
+A Toeplitz (or diagonal-constant) matrix is a matrix in which each descending diagonal
+from left to right is constant, i.e., all elements in a diagonal are same.
+So, Starting from [0][0], Here We will check Elements one by one but next element will be always
+in the next row and next column. if its not found as previous element then its not following Torplitz
+Algorithm so, Return False else if It runs upto end with no retuen means Its Following Toeplitz algo,
+So Return True.
+ */
+package tmatrix;
```
Comment (rish-singhal) [inline Code/Java/Toeplitz.java:11]: add extra line here

File: Code/Java/Toeplitz.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+public class CheckToePlitxMatrix {
+    public static void main(String[] args){
+        Scanner scan = new Scanner(System.in);
+        //Taking Size of Matrix
+        int l = scan.nextInt();
+        int b = scan.nextInt();
+        int[][] matrix = new int[l][b];
+        //Taking Matrix Values
+        for (int i = 0;i<l;i++){
```
Comment (rish-singhal) [inline Code/Java/Toeplitz.java:22]: give proper spaces after ";"

File: Code/Java/Toeplitz.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            }
+        }
+        System.out.println(isToeplitzMatrix(matrix));
+    }
+    public static boolean isToeplitzMatrix(int[][] matrix) {
+        int rlen = matrix.length;
+        int clen = matrix[0].length;
+
+        //horizontal checking
+        for (int i = 0;i<clen;i++){
```
Comment (rish-singhal) [inline Code/Java/Toeplitz.java:34]: here too

File: Code/Java/Toeplitz.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            //temp is count so that j will move till till max row length
+            int temp = 0;
+            int val = matrix[temp][i];
+            temp++;
+            for (int j = i+1;j<clen && temp<rlen;j++){
```
Comment (rish-singhal) [inline Code/Java/Toeplitz.java:39]: here too, and other places

File: Code/Java/Toeplitz.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        }
+
+        //vertical checking
+        for (int i = 1;i<rlen;i++){
+            //temp is count so that j will move till till max column length
+            int temp = 0;
+            int val = matrix[i][temp];
+            temp++;
+            for (int j = i+1;j<rlen && temp<clen;j++){
+                if (val!=matrix[j][temp]){
```
Comment (rish-singhal) [inline Code/Java/Toeplitz.java:54]: and here,

File: Code/Java/Toeplitz.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+               9 5 1 2
+        Output: true
+
+        Input: 2
+               2
+               1 2
+               2 2
+        Output: false
+
+        Time Complexity: O(n^2)
```
Comment (rish-singhal) [inline Code/Java/Toeplitz.java:78]: describe what n is

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by krishna-NIT** [comment]
Thanks for Reviewing @rish-singhal , And From Now Onwards I will Keep this Habit of giving Space, sorry for that, Thank You !

**Comment by krishna-NIT** [comment]
> Thanks for Reviewing @rish-singhal , And From Now Onwards I will Keep this Habit of giving Space, sorry for that, Thank You !

@rish-singhal Changes done !

File: Code/Java/Toeplitz.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        int clen = matrix[0].length;
+
+        //horizontal checking
+        for (int i = 0; i<clen; i++){
+            //temp is count so that j will move till till max row length
+            int temp = 0;
+            int val = matrix[temp][i];
+            temp++;
+            for (int j = i+1; j<clen && temp<rlen; j++){
+                if (val!=matrix[temp][j]){
```
[the remark this answers is not in the document]
Comment (rish-singhal) [reply Code/Java/Toeplitz.java:46]: can you update these?

File: Code/Java/Toeplitz.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+            }
+        }
+        System.out.println(isToeplitzMatrix(matrix));
+    }
+    public static boolean isToeplitzMatrix(int[][] matrix) {
+        int rlen = matrix.length;
+        int clen = matrix[0].length;
+
+        //horizontal checking
+        for (int i = 0; i<clen; i++){
```
Comment (rish-singhal) [inline Code/Java/Toeplitz.java:40]: here too 
around "<"

File: Code/Java/Toeplitz.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+A Toeplitz (or diagonal-constant) matrix is a matrix in which each descending diagonal
+from left to right is constant, i.e., all elements in a diagonal are same.
+So, Starting from [0][0], Here We will check Elements one by one but next element will be always
+in the next row and next column. if its not found as previous element then its not following Torplitz
+Algorithm so, Return False else if It runs upto end with no retuen means Its Following Toeplitz algo,
+So Return True.
+
+*/
+
+
```
Comment (rish-singhal) [inline Code/Java/Toeplitz.java:14]: remove extra lines

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by krishna-NIT** [comment]
@rish-singhal changes made successfully, please review it

**Comment by krishna-NIT** [comment]
@rish-singhal Sir
Please See Discord Once!

**Pull request merged**